packages/core/strapi/lib/services/metrics/index.js

```diff
@@ -79,9 +80,9 @@ const createTelemetryInstance = (strapi) => {
       // clear open handles
       crons.forEach((cron) => cron.cancel());
     },
-    async send(event, payload) {
+    async send(adminUserId, event, payload) {
```

Comment:
It would make the code a more readable to keep the previous format and accept adminUserId in the payload instead. That will also avoid any empty strange calls with `send('', 'event')`

Reply:
I tried that, but then in the end I will have `adminUserId` twice on the payload to the tracker. It's going to be on the payload, which becomes `properties` later, so it will be on properties. And it also needs to be at the top of the body. You suggesting that I delete the key from `properties` object when it gets to the sender?

Reply:
I agree with Alex here.

Honestly, it's a good opportunity right now so I would go a bit further and rethink this interface and the tracker API, you can say no but as we're making those changes and most probably doing a v2 endpoint on the tracker or something similar it's a good window of opportunity.

This function prototype, I added typings for clarity:

```ts
interface Payload {
  userId?: string
  eventProperties?: Record<string, unknown>
  userProperties?: Record<string, unknown>
  groupProperties?: Record<string, unknown>
}

// Not sure we use this one anyway, we could remove it
interface Options {
  // fetch options here
}

function send(event: string, payload?: Payload, options?: Options);
```

With that interface we won't have to update the tracker everytime we want to add `groupProperties` or `userProperties`, more flexibility - and yes more responsponsability - for the API consumer, us.
And also we won't have the issues you mentioned above.

Reply:
@fdel-car's proposal looks pretty good.